airbyte-commons-worker/src/main/java/io/airbyte/workers/internal/DefaultAirbyteStreamFactory.java

```diff
@@ -38,21 +44,39 @@ public class DefaultAirbyteStreamFactory implements AirbyteStreamFactory {
   private final MdcScope.Builder containerLogMdcBuilder;
   private final AirbyteProtocolPredicate protocolValidator;
   protected final Logger logger;
+  private final long maxMemory;
+  private final Optional<Class<? extends RuntimeException>> exceptionClass;
 
   public DefaultAirbyteStreamFactory() {
     this(MdcScope.DEFAULT_BUILDER);
   }
 
   public DefaultAirbyteStreamFactory(final MdcScope.Builder containerLogMdcBuilder) {
-    this(new AirbyteProtocolPredicate(), LOGGER, containerLogMdcBuilder);
+    this(new AirbyteProtocolPredicate(), LOGGER, containerLogMdcBuilder, Optional.empty());
   }
 
   DefaultAirbyteStreamFactory(final AirbyteProtocolPredicate protocolPredicate,
                               final Logger logger,
-                              final MdcScope.Builder containerLogMdcBuilder) {
+                              final MdcScope.Builder containerLogMdcBuilder,
+                              final Optional<Class<? extends RuntimeException>> exceptionClass) {
```

Comment:
Can we add some comments about why we have this `Optional<RuntimeException>`?
Also, this API feels a bit unusual to use as it requires the client to tell what exception to throw. Would throwing a specific exception and let the client code handle the exception wrapping if needed work?